cas-client-core/src/main/java/org/jasig/cas/client/validation/Cas20ProxyReceivingTicketValidationFilter.java

```diff
@@ -139,6 +146,21 @@ public void init() {
         return (T) ReflectUtils.newInstance(ticketValidatorClass, casServerUrlPrefix);
     }
 
+    private PrivateKey buildPrivateKey(final String key) {
+        if (key != null) {
+            final String content = key.replaceAll("\\n", "").replaceAll("\\t", "")
```

Comment:
This is not a good idea. We are assuming keys are PEM files; binary keys should be supported too. Perhaps switch to Bouncycastle and use that to load/parse keys to handle all scenarios?

Reply:
As this private key is meant to be loaded as a filter parameter, I think it makes sense to only accept PEM files (and not binary ones). Here is an example of my local configuration:

```xml
	<filter>
		<filter-name>CAS Validation Filter</filter-name>
		<filter-class>org.jasig.cas.client.validation.Cas30ProxyReceivingTicketValidationFilter</filter-class>
		<init-param>
			<param-name>casServerUrlPrefix</param-name>
			<param-value>http://localhost:8080/cas</param-value>
		</init-param>
		<init-param>
			<param-name>serverName</param-name>
			<param-value>http://localhost:8081</param-value>
		</init-param>
		<init-param>
			<param-name>proxyReceptorUrl</param-name>
			<param-value>/casCallback</param-value>
		</init-param>
		<init-param>
			<param-name>proxyCallbackUrl</param-name>
			<param-value>http://localhost:8081/casCallback</param-value>
		</init-param>
		<init-param>
			<param-name>privateKey</param-name>
			<param-value>
				-----BEGIN PRIVATE KEY-----
				MIICeAIBADANBgkqhkiG9w0BAQEFAASCAmIwggJeAgEAAoGBANy6npifj4rfcDuP
				J2jlha4NhugGVrrmjhuKSRIzEK9dYC8PAv9DLr2LirvP+dmj87ZeBVCV8oVyJdJK
				Xip9aPRHlcgsLP+DxuPAbev20wlQUaEaWyZIJNtnj4ZyOfFEQTy27fHssGGYgdn3
				8W5K3Ce4YgFePiI2z0r+aZg880B/AgMBAAECgYEAuzbGAamENEOji8C1qnDSW3g5
				rl6B/XBDo/Q8WeVaB4Npa0COSDeRU08h0n5p6kD3GvzQATPNSYYxshVFnUBVXRgH
				YwE2XUWYXCMkuxnsT0lsPzhBBZQBl+Z2niF16y/M8kD4WtU25jFUnyw7ymY6/a2i
				56p4lgw6VKMWxB9HzRECQQD5tbERNTxFHbJsFbiwgmNFdv9KvO2lGqNIKepwVhrf
				xCtphxZ+T9MZT+7xTmPf3EjZQzyk1KvyP6WLlkL5JjOTAkEA4koKAS4YidLwHkNW
				G4OzB/RpGyfs/eaLryQBmpRoj/tnnlqil6S6yDTaor1afRoRJF2Ns27SA61r2ul6
				1gsq5QJBAMgT87dE/3CltdNyqBQNtTrweOdKb+ONb20f4K933nqStPIOAlDc5tju
				Ohbw54dz+o7Sl+G+YbNtmdg5Ypo9KK8CQD6xfpYXHEtK+qWXyEp1MWu5xkj0xc8O
				SZLradpW1aVWZL3VpIhY5jrDkfNWfLR+PVOS68hTorHMIT6UQyj8DRUCQQC2idIA
				FbF3gID9A5ZuL1kqk9QzVPmj/IBOF5LwNORkoUDZcNcHkCPWUKNj/lBo6F/ITYCe
				+W6QJE/M7kVYdpsp
				-----END PRIVATE KEY-----
			</param-value>
		</init-param>
	</filter>
	<filter-mapping>
		<filter-name>CAS Validation Filter</filter-name>
		<url-pattern>*</url-pattern>
	</filter-mapping>
```

Reply:
Wouldn't also want to accept a path to a file instead? I'd say that's way more common than pasting the key contents here, and you'd cover more ground. (Copy/paste and formatting is also prone to error) 

I think the docs also start out with DER files as an example on how to generate keys: https://apereo.github.io/cas/development/installation/Configuring-Proxy-Authentication.html#register-service

Reply:
Something like this, per @dima767 would also have a difficult time configuring the key contents:
https://github.com/Unicon/cas-client-autoconfig-support

Reply:
OK. It's done. I now use bouncycastle and it works like in the private key factory bean of the CAS server. What do you think?